pipeline-steps/deploy.groovy

```diff
@@ -0,0 +1,19 @@
+def deploy(){
+  common.conditionalStep(
+    step_name: "Deploy",
+    step: {
```

Comment:
I added a quick function for this, but it occurred to me that running deploy.sh is going to repeat most of the prep stage. There are conditionals in the deploy script that allow prep steps only but no actual deploys, but not the other way round.

The majority of the deploy script is tested in the aio prep stage however: https://github.com/rcbops/rpc-gating/blob/bug/1069/pipeline-steps/aio_prepare.groovy#L14